gdx/src/com/badlogic/gdx/math/Quaternion.java

```diff
@@ -780,7 +792,7 @@ public float getAxisAngleRad (Vector3 axis) {
 		float angle = (float)(2.0 * Math.acos(this.w));
 		double s = Math.sqrt(1 - this.w * this.w); // assuming quaternion normalised then w is less than 1, so term always positive.
 		if (s < MathUtils.FLOAT_ROUNDING_ERROR) { // test to avoid divide by zero, s is always positive due to sqrt
-			// if s close to zero then direction of axis not important
+																// if s close to zero then direction of axis not important
```

Comment:
These indented comments only align when using a tab size of 3, which is unusual. I expect most people here use a size of 4 or 2, and GitHub defaults to 8 for some insane reason. It's unfortunate we don't all use the same tab size, as I am also a fan of aligning things this way (just I use spaces to do so if it's my own codebase).

Sorry I can't add anything particularly insightful to this pull request. Static primitives are okay - it's just the objects that are problematic?

I'm neutral on these changes if they don't break anything or greatly affect performance (a small performance impact I'd never notice and some of libGDX's optimisations are no longer important anyway) but I think it's most likely this will have to remain as a fork.

Reply:
Static primitives still have issues with multiple threads, if there's contention for that static variable.

Reply:
Most of the mods in this PR is change static fields to class instance fields.
This was easy, as there is really no need to make them static. No perf impact, better data design.
As @tommyettinger says, static primitives may have issues, so we basically checked them without finding any real danger.
We did not cleanup all the code with overrides and finals and visibility changes ofc. The result would have been a gigantic PR .. impossible to handle.

Reply:
Okay, sounds good.

Reply:
For our projects we are using Eclipse whose code cleanup is quite efficient and accurate.
So we're no longer bothering with self-made formatting, finals etc. We do format on save action.

BTW 'final' is crucial for a better design and.. gives some spice to the compiler optimization.